packages/website/src/components/ErrorsViewer.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import type Monaco from 'monaco-editor';
+import type { ErrorItem } from './types';
+
+import styles from './ErrorsViewer.module.css';
+
+export interface ErrorsViewerProps {
+  readonly value?: ErrorItem[];
+}
+
+export interface ErrorBlockProps {
+  readonly item: ErrorItem;
+  readonly onClick: () => void;
+  readonly isLocked: boolean;
+}
+
+function severityClass(severity: Monaco.MarkerSeverity): string {
+  switch (severity) {
+    case 8:
+      return 'danger';
+    case 4:
+      return 'caution';
+    case 2:
+      return 'note';
+  }
+  return 'info';
+}
+
+function ErrorBlock({ item, onClick, isLocked }: ErrorBlockProps): JSX.Element {
+  return (
+    <div className={`admonition alert alert--${severityClass(item.severity)}`}>
+      <div className="admonition-content">
+        <div className="row row--no-gutters">
+          <div className="col col--12">
+            {item.message} {item.location}
+          </div>
+          {item.hasFixers && (
+            <div className="col col--12">
+              {item.fixers.map((fixer, index) => (
+                <div key={index} className={styles.fixer}>
+                  <span>&gt; {fixer.message}</span>
+                  <button
+                    className="button button--primary button--sm"
+                    disabled={isLocked}
+                    onClick={(): void => {
+                      fixer.fix();
+                      onClick();
+                    }}
+                  >
+                    fix
+                  </button>
+                </div>
+              ))}
+            </div>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default function ErrorsViewer({
+  value,
+}: ErrorsViewerProps): JSX.Element {
+  const [model, setModel] = useState<[string, ErrorItem[]][]>();
+
+  const [isLocked, setIsLocked] = useState<boolean>(false);
+
+  const onClick = useCallback(() => {
+    setIsLocked(true);
+  }, [isLocked]);
```

Comment:
This is unnecessary - the state setters are always constant, irrespective of their state value.

```suggestion
  }, []);
```

Reply:
i missed that after refactoring